### PR TITLE
DOCS-492: Container Platform Docs

### PR DESCRIPTION
**Pull request by ravindk89**
# Summary

This commit adds a `make container` build for Docker/Podman

This commit also generally cleans up the Installation docs and uses /includes/<platform>/steps-*.rst as a pattern for storing platform-specific tutorials within a single root URL.

Finally, due to API rate limits, this commit adds some Makefile magic to allow skipping of sync'ing dependencies. Pass `make BUILD_DEPENDENCIES=FALSE <platform>` to skip building any deps and use what is already in the branch.

# Notes

This PR will require a refresh once my API restrictions fall off such that the various SDK docs are not negatively impacted. Please ignore the massive swath of deletions.

No staging links available yet, as my local builds are slightly broken. Will update once the API limit drops.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 # You can set these variables from the command line, and also
 # from the environment for the first two.
-SPHINXOPTS    ?= -n -j "auto"
```
Comment (ravindk89) [inline Makefile:6]: Logs output to a text file so it's easier to track what has gone wrong. Already excluded from git.

File: Makefile [hunk 2]
```diff
@@ -35,35 +35,63 @@ stage-%:
 	@echo "Visit http://localhost:8000 to view the staged output"
 
 linux:
+ifeq ($(BUILD_SDK),FALSE)
```
Comment (ravindk89) [inline Makefile:38]: Typo, should be `BUILD_DEPENDENCIES`, will fix on next commit pass.

File: source/default-conf.py [hunk 3]
```diff
@@ -173,6 +173,32 @@
...
+    excludes = [
+        'operations/install-deploy-manage/deploy-minio-tenant.rst',
+        'operations/install-deploy-manage/modify-minio-tenant.rst',
+        'operations/install-deploy-manage/expand-minio-tenant.rst',
+        'operations/install-deploy-manage/upgrade-minio-tenant.rst',
+        'operations/install-deploy-manage/upgrade-minio-operator.rst',
+        'operations/install-deploy-manage/delete-minio-tenant.rst',
+        'operations/install-deploy-manage/minio-operator-console.rst',
+        'operations/install-deploy-manage/deploy-minio-multi-node-multi-drive.rst',
+        'operations/install-deploy-manage/multi-site-replication.rst',
```
Comment (ravindk89) [inline source/default-conf.py:186]: Intentionally excluded in this pass.

- Distributed deploys via containers should be done via K8s. We will not be publishing docs for compose, swarm, helm, etc.
- Multi-Site can be handled in a separate pass if at all, as container networking makes this very finicky.

File: source/operations/install-deploy-manage/deploy-minio-single-node-multi-drive.rst [hunk 4]
```diff
@@ -59,312 +62,14 @@ deployments are best suited for evaluation and initial development environments.
...
-      :start-after: start-run-minio-binary-desc
-      :end-before: end-run-minio-binary-desc
-
-.. cond:: macos
-
-   .. include:: /includes/macos/common-installation.rst
-      :start-after: start-run-minio-binary-desc
-      :end-before: end-run-minio-binary-desc
-
-3) Add TLS Certificates
```
Comment (ravindk89) [inline source/operations/install-deploy-manage/deploy-minio-single-node-multi-drive.rst:92]: Intentionally dropped, as the dedicated TLS tutorial page can provide better instructions. If we want to go this route, it should be as part of a polish pass where the deploy tutorials cover *all* possible setups, including TLS, KES, etc. That requires significantly more effort and care than we have bandwidth for today.

File: source/includes/macos/steps-deploy-minio-single-node-single-drive.rst [hunk 5]
```diff
@@ -0,0 +1,86 @@
+1) Download the MinIO Server
```
Comment (djwfyi) [inline source/includes/macos/steps-deploy-minio-single-node-single-drive.rst:1]: Not going to relist them, but this needs the same fixes as the other files.

File: source/operations/install-deploy-manage/deploy-minio-single-node-multi-drive.rst [hunk 6]
```diff
@@ -59,312 +62,14 @@ deployments are best suited for evaluation and initial development environments.
...
-``https://127.0.0.1:9090``.
-
-Log in with the :guilabel:`MINIO_ROOT_USER` and :guilabel:`MINIO_ROOT_PASSWORD`
-from the previous step.
-
-.. image:: /images/minio-console/minio-console.png
-   :width: 600px
-   :alt: MinIO Console Dashboard displaying Monitoring Data
-   :align: center
+   .. include:: /includes/macos/steps-deploy-minio-single-node-multi-drive.rst
```
Comment (djwfyi) [inline source/operations/install-deploy-manage/deploy-minio-single-node-multi-drive.rst:71]: Windows content is missing.
Need to add the windows conditional and content.

File: source/operations/install-deploy-manage/deploy-minio-single-node-multi-drive.rst [hunk 7]
```diff
@@ -40,12 +45,10 @@ Local JBOD Storage with Sequential Mounts
...
------------------------------------
+Deploy Single-Node Multi-Drive MinIO
+------------------------------------
 
-The following procedure deploys MinIO in :guilabel:`Standalone Mode` consisting
-of a single MinIO server and a single drive or storage volume. Standalone
-deployments are best suited for evaluation and initial development environments.
+The following procedure deploys MinIO consisting of a single MinIO server and a single drive or storage volume.
 
 .. admonition:: Network File System Volumes Break Consistency Guarantees
```
Comment (djwfyi) [inline source/operations/install-deploy-manage/deploy-minio-single-node-multi-drive.rst:53]: This admonition is really repetitive to the one just above it on the page. (Lines 36-44 are basically the same, and there's just a heading and one line paragraph between them.)

**Review by djwfyi — CHANGES_REQUESTED** [review]
Missing content for Windows in the SNMD page.
Some suggestions for repetitive content that would be great to move into a more single sourced mechanism.

Other bits for spelling, punctuation, or clarity throughout.

I'll take another look after those are done.

File: source/operations/install-deploy-manage/deploy-minio-single-node-single-drive.rst [hunk 8]
```diff
@@ -17,7 +19,7 @@ This feature allows access to :ref:`erasure coding dependent features <minio-era
 MinIO only starts in |SNSD| mode if the storage volume or path is empty *or* only contain files generated by a previous |SNSD| deployment.
 See :ref:`minio-snsd-pre-existing-data <minio-snsd-pre-existing-data>` for more complete documentation on MinIO startup behavior in |SNSD| mode.
```
[the remark this answers is not in the document]
Comment (ravindk89) [reply source/operations/install-deploy-manage/deploy-minio-single-node-single-drive.rst]: Done

**Review by ravindk89 — COMMENTED** [review]
@djwfyi back to you. Significant single sourcing pass. Skipping Windows as I need to verify how that all behaves first.

File: source/includes/container/steps-deploy-minio-single-node-multi-drive.rst [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+   :end-before: end-common-deploy-validate-container-status
+
+.. code-block:: none
+
+   Formatting 1st pool, 1 set(s), 4 drives per set.
+   WARNING: Host local has more than 4 drives of set. A host failure will result in data becoming unavailable.
+   MinIO Object Storage Server
+   Copyright: 2015-2022 MinIO, Inc.
+   License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
+   Version: RELEASE.2022-07-08T00-05-23Z (go1.18.3 linux/amd64)
```
Comment (djwfyi) [inline source/includes/container/steps-deploy-minio-single-node-multi-drive.rst:135]: Is it worth adding a substitution to this code block so you can use `|minio-latest|`?

If so, we'd need to figure out what we're going to do with line 133 each January.

File: source/includes/container/steps-deploy-minio-single-node-single-drive.rst [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+   :end-before: end-common-deploy-validate-container-status
+
+.. code-block:: shell
+
+   Formatting 1st pool, 1 set(s), 1 drives per set.
+   WARNING: Host local has more than 0 drives of set. A host failure will result in data becoming unavailable.
+   MinIO Object Storage Server
+   Copyright: 2015-2022 MinIO, Inc.
+   License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
+   Version: RELEASE.2022-07-08T00-05-23Z (go1.18.3 linux/amd64)
```
Comment (djwfyi) [inline source/includes/container/steps-deploy-minio-single-node-single-drive.rst:132]: Same question here. Use the global variable from `conf.py`?
Need to consider what happens to line 130 each January.

File: source/includes/linux/steps-deploy-minio-single-node-multi-drive.rst [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+The ``journalctl`` output should resemble the following:
+
+.. code-block:: shell
+
+   Formatting 1st pool, 1 set(s), 1 drives per set.
+   WARNING: Host local has more than 0 drives of set. A host failure will result in data becoming unavailable.
+   MinIO Object Storage Server
+   Copyright: 2015-2022 MinIO, Inc.
+   License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
+   Version: RELEASE.2022-07-08T00-05-23Z (go1.18.3 linux/amd64)
```
Comment (djwfyi) [inline source/includes/linux/steps-deploy-minio-single-node-multi-drive.rst:40]: Variables?
Line 38?

File: source/includes/linux/steps-deploy-minio-single-node-single-drive.rst [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+The ``journalctl`` output should resemble the following:
+
+.. code-block:: shell
+
+   Formatting 1st pool, 1 set(s), 1 drives per set.
+   WARNING: Host local has more than 0 drives of set. A host failure will result in data becoming unavailable.
+   MinIO Object Storage Server
+   Copyright: 2015-2022 MinIO, Inc.
+   License: GNU AGPLv3 <https://www.gnu.org/licenses/agpl-3.0.html>
+   Version: RELEASE.2022-07-08T00-05-23Z (go1.18.3 linux/amd64)
```
Comment (djwfyi) [inline source/includes/linux/steps-deploy-minio-single-node-single-drive.rst:40]: Variable here?
Line 38?

File: source/operations/install-deploy-manage/deploy-minio-single-node-single-drive.rst [hunk 13]
```diff
@@ -17,7 +19,7 @@ This feature allows access to :ref:`erasure coding dependent features <minio-era
...
 
-For extended development or production environments, deploy MinIO in :guilabel:`Distributed Mode`. See :ref:`deploy-minio-distributed` for more information.
+For extended development or production environments, deploy MinIO in :guilabel:`Distributed Mode`. See :ref:`deploy-minio-distributed` for mo/operations/install-deploy-manage/multi-site-replicationre information.
```
[the remark this answers is not in the document]
Comment (djwfyi) [reply source/operations/install-deploy-manage/deploy-minio-single-node-single-drive.rst:22]: Need to fix this paste. The path is not in the :ref: link and splitting the word `more` instead.

**Review by djwfyi — CHANGES_REQUESTED** [review]
Minor nits.
One bad path that looks like a possible paste error.

Several of the comments are about using a variable in example output so that the file stays up-to-date automatically. But then we have to think about the copyright year line in each case.

Is there an issue for adding the Windows content?

Other than that, this LGTM after those fixes.

**Review by djwfyi — APPROVED** [review]
LGTM

**Pull request merged**